### PR TITLE
Add speaking pages

### PR DESCRIPTION
**Pull request by dancork**
## What does this PR do?
Adds a speaking page and profile pages to hungry geek. Very basic atm, limited data but folks can put PRs in for their own talks and profiles :)

## How to test
- Checkout branch
- Run `jekyll serve --config _config.yml,_config_dev.yml`
- Visit `http://localhost:4000/speaking/`

#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've updated documentation with any changes to procedures..
- [x] I am ready for this to be code reviewed, merged and tested.

#### IP or Developer Review:
- [ ] I’ve witnessed the work behaving as expected.
- [ ] I’ve checked for coding anti-patterns.
- [ ] +1 from me!

#### Software Engineer, Software Architect or Developer review:
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for coding anti-patterns.
- [x] +1 from me!

### CODE REVIEW

**Comment by hpoom** [comment]
:+1:

**Comment by jodiedoubleday** [comment]
Can we merge members.yml and speakers? rather than having a .md page for each speaker?

File: _includes/head.html [hunk 1]
```diff
@@ -42,6 +42,6 @@
     <meta name="twitter:creator:id" "">
     -->
 
-    <style>{% include styles.html %}</style>
+    <style>{% include styles.css %}</style>
```
Comment (jodiedoubleday) [inline _includes/head.html]: Happy for you to change this to a css file (nobody likes it as html) but we need to ensure the css is deferrered as not to affect our page speed optimisation
`https://developers.google.com/speed/pagespeed/insights/?url=hungrygeek.holidayextras.co.uk`

**Comment by dancork** [comment]
@jodiedoubleday I considered using the yml but thought we'd want to make to the profile pages a little more individual, possibly with subpages for some of the talks - with custom text, video, slides etc. Would help with future CFP submissions.

Probably better to keep the yml clean and light too. Happy to change though if others disagree.

File: _includes/head.html (hunk 1 above)
Comment (dancork) [inline _includes/head.html]: I believe it acted in the same way, will need to double check as this was at 1am!

Tempted to add grunt/gulp so we can use less. Then exclude the grunt/gulp file and less dir in the Jekyll config. Seen it in a few Jekyll themes. What do you think? Will do this in a separate PR though.

File: _includes/styles.css [hunk 2]
```diff
@@ -387,6 +396,116 @@
...
+
+/* --- Speaking --- */
+.speaking section {
+  margin: 2.5em 0;
+}
+.speaking h2 {
+  text-align: center;
+  font-size: 2.2em;
+}
+.speaking-calendar {
```
Comment (lukefrake) [inline _includes/styles.css]: Couple of empty selectors in this file

File: _data/speaking.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+- name: 'TEDx University of Kent'
+  location: 'Canterbury, UK'
+  link: 'http://www.tedxuniversityofkent.com/'
+  pretty_date: '30th May, 2015'
+  date_for_ordering: '2015-05-30'
+  month: '5'
+  year: '2015'
+  speakers:
+    - name: 'Luke Frake'
+
```
Comment (lukefrake) [inline _data/speaking.yml]: - talk: 'The internet needs us, all of us.'

File: _includes/speaker_list.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{% if speakers.size %}
```
Comment (lukefrake) [inline _includes/speaker_list.html]: Not sure what your rules are on indention, but I think some files should be more regularly indented?

**Comment by lukefrake** [comment]
So good to see progress on this!

File: _includes/head.html (hunk 1 above)
Comment (lukefrake) [inline _includes/head.html]: Personally, I wouldn't get too hung up on grunt/gulp as part of this PR, the PR is very good progress on it's own.

In terms of the file being deferred, obviously inline it wouldn't have been, so again would say this is progress in the right direction?

There are other bigger optimisations we could do, having this in a CSS file is an optimisation in itself, because it can be cached, where as inlined it's downloaded every time.

What do you think @jodiedoubleday ?

**Comment by dancork** [comment]
@lukefrake made a few amends :)

File: _includes/head.html (hunk 1 above)
Comment (dancork) [inline _includes/head.html]: Could use something like this in the footer, seems to work...

```
<script>
  function loadStyleSheet(src){
    if (document.createStyleSheet) document.createStyleSheet(src);
    else {
      var stylesheet = document.createElement('link');
      stylesheet.href = src;
      stylesheet.rel = 'stylesheet';
      stylesheet.type = 'text/css';
      document.getElementsByTagName('head')[0].appendChild(stylesheet);
    }
  }
  (function(){ loadStyleSheet('{{site.url}}{{site.base_url}}/css/styles.css'); }());
</script>
```

Comment (lukefrake) [inline _includes/head.html]: :+1: this is what we use on SB sites for loading in JS

**Comment by dancork** [comment]
@jodiedoubleday @lukefrake have moved css to the footer with js script to post load.

will look into implementing grunt/gulp, less and better mobile perf in a different pr.

**Comment by lukefrake** [comment]
:+1: from me, as long as @jodiedoubleday is happy with the members & speakers yml files

**Comment by jodiedoubleday** [comment]
Great to see the css moved to the bottom with JS.
Still not happy with the duplication between individual md files per speaker and the member.yml. Id say push the speakers info into members. Have one plce for gravatars, twitter handles etc

**Comment by dancork** [comment]
Spoken to @jodiedoubleday, going to combine speaking profiles and the team page. For this PR moving them from `speaking/` to `profiles/`. Will change the team page at a later date.

**Comment by hpoom** [comment]
I like the idea of speaking profiles and team page merging into one.
